server/assetstore.py

```diff
@@ -34,6 +34,43 @@
 DB_INFO_KEY = 'databaseMetadata'
 
 
+class DatabaseAssetstoreFile(dict):
+    """
+    This wraps a Girder file object dictionary so that the size parameter can
+    be determined only when actually necessary.
+    """
+    def __init__(self, file, adapter, *args, **kwargs):
+        """
+        Create a file dictionary where 'size' and 'mimeType' are determined
+        lazily.
+
+        :param file: the original file dictionary.
+        :param adapter: the owning assetstore adapter.
+        """
+        self._file = file
+        self._adapter = adapter
+        self._loaded = None
+        return super(DatabaseAssetstoreFile, self).__init__(file, *args, **kwargs)
+
+    def __getitem__(self, key, *args, **kwargs):
```

Comment:
What is the reasoning for using __getitem__ over @property for lazily evaluated properties?